grails-app/controllers/org/pih/warehouse/api/InternalLocationApiController.groovy

```diff
@@ -44,27 +36,19 @@ class InternalLocationApiController {
         render([data: locations, totalCount: locations?.totalCount] as JSON)
     }
 
-    def listReceiving() {
-        String locationId = params?.location?.id ?: session?.warehouse?.id
-        Location parentLocation = locationId ? Location.get(locationId) : null
-        if (!parentLocation) {
-            throw new IllegalArgumentException("Must provide location.id as a request parameter")
-        }
-        String shipmentNumber = params?.shipmentNumber
-        if (!shipmentNumber) {
-            throw new IllegalArgumentException("Must provide shipmentNumber as a request parameter")
-        }
-
-        LocationTypeCode[] locationTypeCodes = params.locationTypeCode ? params.list("locationTypeCode") : [LocationTypeCode.BIN_LOCATION]
+    def listReceiving(ReceivingBinSearchCommand command) {
+        command.location = getFacility(command)
 
-        Map activityCodeParams = [:]
-        activityCodeParams.allActivityCodes = params.allActivityCodes ? params.list("allActivityCodes") as ActivityCode[]: []
-        activityCodeParams.anyActivityCodes = params.anyActivityCodes ? params.list("anyActivityCodes") as ActivityCode[]: []
-        activityCodeParams.ignoreActivityCodes = params.ignoreActivityCodes ? params.list("ignoreActivityCodes") as ActivityCode[] : [ActivityCode.RECEIVE_STOCK]
-
-        String[] receivingLocationNames = [locationService.getReceivingLocationName(shipmentNumber), "Receiving ${shipmentNumber}"]
-        List<Location> locations = locationService.getInternalLocations(parentLocation, locationTypeCodes, activityCodeParams, receivingLocationNames)
+        // When searching receiving bins, we need different default search parameters. Mainly we want to include
+        // the normal bin locations as well as only the receiving bins that match the given shipment number.
+        command.locationTypeCode = params.locationTypeCode ? params.list("locationTypeCode") : [LocationTypeCode.BIN_LOCATION]
```

Comment:
Something weird is going on here. It's throwing the following:
```
Java.lang.String cannot be cast to java.lang.Enum
```
when the query in LocationService runs:
```
List<Location> internalLocations = Location.createCriteria().list() {
    locationType {
        'in'("locationTypeCode", command.locationTypeCode)
    }
}
```
For some reason it works totally fine for cycle count and command.locationTypeCode is exactly the same in both those scenarios so I don't know what's going on. I'm still looking into it